server/src/main/java/io/druid/segment/realtime/firehose/EventReceiverFirehoseFactory.java

```diff
@@ -207,8 +197,11 @@ public boolean hasMore()
     {
       synchronized (readLock) {
         try {
-          while (!closed && nextRow == null) {
+          while (nextRow == null) {
             nextRow = buffer.poll(500, TimeUnit.MILLISECONDS);
+            if (closed) {
```

Comment:
This lets `hasMore` keep returning true post-close until the buffer is empty (poll returns null after the 500ms timeout)